crypto/fips1402/src/main/java/org/keycloak/crypto/fips/FIPS1402Provider.java

```diff
@@ -262,4 +259,34 @@ private SNIHostName getSNIHostName(String host) {
 
         };
     }
+
+    // BCFIPS require "SecureRandom.getInstanceStrong" to be available. But it may not be available on RHEL 8 on OpenJDK 17 due the https://bugzilla.redhat.com/show_bug.cgi?id=2155060
+    private void checkSecureRandom(Runnable insertBcFipsProvider) {
+        try {
+            SecureRandom sr = SecureRandom.getInstanceStrong();
+            log.debugf("Strong secure random available. Algorithm: %s, Provider: %s", sr.getAlgorithm(), sr.getProvider());
+            insertBcFipsProvider.run();
+        } catch (NoSuchAlgorithmException nsae) {
+
+            // Fallback to regular SecureRandom
+            SecureRandom secRandom = new SecureRandom();
+            String origStrongAlgs = Security.getProperty("securerandom.strongAlgorithms");
```

Comment:
Is this property not set by any chance? If you look at the `SecureRandom` impl it should always fail with `NoSuchAlgorithmException`  when calling `getInstanceStrong` if the property is not set.

Reply:
There is also another possibility how the `NoSuchAlgorithmException` can be thrown from the `getInstanceStrong` - when property `"securerandom.strongAlgorithms` is specified, but it's pointing to the algorithms, which are not available or disabled. That is the case here.

This property is set by default in `java.security` on all the environments I saw so far. And it is typically set in OpenJDK by default to `NativePRNGBlocking:SUN,DRBG:SUN` (applies for both my Ubuntu as well as for RHEL with or without FIPS enabled, and for both OpenJDK11 and OpenJDK17). The problem is that none of the specified algorithms `NativePRNGBlocking:SUN,DRBG:SUN` is available on the OpenJDK 17 on the FIPS enabled RHEL. That's also clear fro m the exception message (For example see here how the exception looks like: https://bugzilla.redhat.com/show_bug.cgi?id=2155060)

I think that it is a bug as `getInstanceStrong` should be available per javadoc of this method. The `SecureRandom.getInstanceStrong()` is used by BCFIPS and BCFIPS fails to bootstrap when this method is not available. Also I talked about this with the guy from BouncyCastle team and he confirmed that `getInstanceStrong` unavailable looks like a bug (and suggested to use exactly the workaround like this).

You can check https://bugzilla.redhat.com/show_bug.cgi?id=2155060 for the details (I don't yet have response from the OpenJDK team here).